src/attr_file.c

```diff
@@ -397,7 +397,7 @@ bool git_attr_fnmatch__match(
 
 		/* for attribute checks or root ignore checks, fail match */
 		if (!(match->flags & GIT_ATTR_FNMATCH_IGNORE) ||
-			path->basename == path->path)
+			path->basename == relpath)
```

Comment:
That line of thinking is what I was thinking. :)

I fixed and force-pushed the changes requested w.r.t. test code style, and an attempt at updating the comment here.